obsei/preprocessor/text_cleaner.py

```diff
@@ -0,0 +1,177 @@
+import logging
+from typing import Any, Dict, List, Optional
+import string
+import re
+from unicodedata import normalize as unicode_decode
+from dateutil.parser import parse
+
+import nltk
+from nltk.corpus import stopwords
+from nltk.stem import PorterStemmer
+from nltk.tokenize import word_tokenize
+
+from pydantic import PrivateAttr
+
+from obsei.analyzer.base_analyzer import AnalyzerRequest
+from obsei.preprocessor.base_text_cleaner import (
+    BaseTextCleaner,
+    BaseTextCleanerConfig,
+)
+
+logger = logging.getLogger(__name__)
+nltk.download("punkt")
+
+
+class TextCleaner(BaseTextCleaner):
+    text_cleaning_functions: List = []
+
+    def __init__(self, **data: Any):
+        super().__init__(**data)
+        self.stop_words: List = data.get(
+            "stop_words", stopwords.words(data.get("language", "english"))
```

Comment:
Language should be config parameter of TextCleanerConfig.
I can explain design principal about abstraction of Config an ActualComponent class.
Main idea is to split stateful information from operation. Hence config class will keep any dynamic pieces and Component like TextCleaner will be always stateless.